lwt/cohttp_lwt.ml

```diff
@@ -298,7 +298,14 @@ module Make_server(IO:IO)
     Filename.concat docroot rel_path
 
   let respond ?headers ?(flush=true) ~status ~body () =
-    let encoding = Cohttp_lwt_body.transfer_encoding body in
```

Comment:
That's a very good point though. There's quite a bit of subtle logic that could be pulled into Cohttp itself.  Am putting together an Async proxy now